> When I checkout this branch, I get this error when building:
> 
> ```
> Running phase: buildPhase
> [webpack-cli] Failed to load '/build/source/frontend/webpack.config.js' config
> [webpack-cli] Error: Cannot find module 'webpack'
> Require stack:
> - /build/source/frontend/config/webpack.config.prod.js
> - /build/source/frontend/webpack.config.js
> - /nix/store/37q963ifg5v5k8r9yn8g165pgp5g6fhs-webpack-cli-5.1.4/lib/node_modules/webpack-cli/lib/webpack-cli.js
> - /nix/store/37q963ifg5v5k8r9yn8g165pgp5g6fhs-webpack-cli-5.1.4/lib/node_modules/webpack-cli/lib/bootstrap.js
> - /nix/store/37q963ifg5v5k8r9yn8g165pgp5g6fhs-webpack-cli-5.1.4/lib/node_modules/webpack-cli/bin/cli.js
>     at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
>     at Function._load (node:internal/modules/cjs/loader:1075:27)
>     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
>     at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
>     at Module.require (node:internal/modules/cjs/loader:1340:12)
>     at require (node:internal/modules/helpers:141:16)
>     at Object.<anonymous> (/build/source/frontend/config/webpack.config.prod.js:3:34)
>     at Module._compile (node:internal/modules/cjs/loader:1546:14)
>     at Object..js (node:internal/modules/cjs/loader:1689:10)
>     at Module.load (node:internal/modules/cjs/loader:1318:32) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/build/source/frontend/config/webpack.config.prod.js',
>     '/build/source/frontend/webpack.config.js',
>     '/nix/store/37q963ifg5v5k8r9yn8g165pgp5g6fhs-webpack-cli-5.1.4/lib/node_modules/webpack-cli/lib/webpack-cli.js',
>     '/nix/store/37q963ifg5v5k8r9yn8g165pgp5g6fhs-webpack-cli-5.1.4/lib/node_modules/webpack-cli/lib/bootstrap.js',
>     '/nix/store/37q963ifg5v5k8r9yn8g165pgp5g6fhs-webpack-cli-5.1.4/lib/node_modules/webpack-cli/bin/cli.js'
>   ]
> }
> error: builder for '/nix/store/023nms1lifgz47i4xl4b72hgprdfrkw0-statping-ng-0.91.0-go-modules.drv' failed with exit code 2;
> ```

thats interesting, it should be using webpack-6.0.1 instead

--- a/pkgs/by-name/st/statping-ng/package.nix
+++ b/pkgs/by-name/st/statping-ng/package.nix
@@ -1,0 +1,104 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  go-rice,
+  lib,
+  nodejs,
+  stdenv,
+  yarnBuildHook,
+  yarnConfigHook,
+}:
+let
+  version = "0.92.0";
+
+  src = fetchFromGitHub {
+    owner = "statping-ng";
+    repo = "statping-ng";
+    tag = "v${version}";
+    hash = "sha256-E4sVIa8sKmjRcduATTHLklkr+LKX6KucDw42uVFhK4g=";
+  };
+
+  frontend = stdenv.mkDerivation {
+    pname = "statping-ng-frontend";
+    inherit version;
+    src = "${src}/frontend";
+
+    yarnOfflineCache = fetchYarnDeps {
+      yarnLock = "${src}/frontend/yarn.lock";
+      hash = "sha256-e8GyKIJ0RopRliVMVrY8eEd6Qx/gTKbW3biPCSqbRrQ=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      yarnConfigHook
+      yarnBuildHook
+    ];
+
+    preBuild = ''
+      export NODE_OPTIONS=--openssl-legacy-provider
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p "$out"
+      cp -rt "$out" dist/* src/assets/scss public/robots.txt
+
+      runHook postInstall
+    '';
+  };
+in
+buildGoModule rec {
+  pname = "statping-ng";
+  inherit version src;
+
+  proxyVendor = true;
+  vendorHash = "sha256-ZcNOI5/Fs7/U8/re89YpJ3qlMaQStLrrNHXiHuBQwQk=";
+
+  postPatch = ''
+    ln -s "${frontend}" source/dist
+  '';
+
+  nativeBuildInputs = [
+    go-rice
+  ];
+
+  preBuild = ''
+    (cd source && rice embed-go)
+  '';
+
+  subPackages = [
+    "cmd/"
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.VERSION=${version}"
+  ];
+
+  tags = [
+    "netgo"
+    "ousergo"
+  ];
+
+  doCheck = false;
+
+  postInstall = ''
+    mv $out/bin/cmd $out/bin/statping-ng
+    $out/bin/statping-ng version | grep ${version} > /dev/null
+  '';
+
+  meta = {
+    description = "Status Page for monitoring your websites and applications with beautiful graphs, analytics, and plugins";
+    homepage = "https://github.com/statping-ng/statping-ng";
+    changelog = "https://github.com/statping-ng/statping-ng/releases/tag/v${src.tag}";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [
+      FKouhai
+    ];
+    platforms = lib.platforms.linux;
+    mainProgram = "statping-ng";
+  };
+}

